cmd/cdi-cloner/cloner_startup.sh

```diff
@@ -33,15 +40,24 @@ if [ "$obj" == "source" ]; then
     echo "cloner: Starting clone source"
     echo "cloner: creating fifo pipe"
     mkfifo $pipe_dir
-    echo "cloner: creating tarball of the image and redirecting it to $pipe_dir"
-    pushd $image_dir
-	#figure out the size of content in the directory
-    size=$(du -sb . | cut -f1)
-    echo $size
-    #Write the size to the pipe so the other end can read it.
-    echo "$size" >$pipe_dir
-    tar cv --sparse . >$pipe_dir
-    popd
+    if [ "$volumeMode" != "block" ]; then
+        echo "cloner: creating tarball of the image and redirecting it to $pipe_dir"
+        pushd $image_dir
+	    #figure out the size of content in the directory
+        size=$(du -sb . | cut -f1)
+        echo " size is"
+        echo $size
+        echo "going to echo size to pipe dir"
+        #Write the size to the pipe so the other end can read it.
+        echo "$size" >$pipe_dir
+        echo "before tar"
+        tar cv --sparse . >$pipe_dir
+        echo "tar completed"
+        popd
+    else
+        echo "cloner: writing the image to $pipe_dir"
+        dd if=$image_dir bs=4M > $pipe_dir
```

Comment:
you can do: dd if=$image_dir bs=4M of=$pipe_dir should save you one shell invocation I think.